File: agent/src/main/java/io/github/vicitori/threading/highlighter/agent/common/MarkerInfo.java
package io.github.vicitori.threading.highlighter.agent.common;

/**
 * Describes a threading assertion marker method to instrument.
 */
public record MarkerInfo(
        String classFqn,
        String methodName,
        String displayName,
        String description
) {
    public String markerFqn() {
        return classFqn + "#" + methodName;
    }
}

File: common/src/main/kotlin/io/github/vicitori/threading/highlighter/common/marker/MarkerInfo.kt
package io.github.vicitori.threading.highlighter.common.marker

data class MarkerInfo(
    @JvmField val classFqn: String,
    @JvmField val methodName: String,
    @JvmField val displayName: String,
    @JvmField val description: String
) {
    fun markerFqn(): String = "$classFqn#$methodName"
}
